homeassistant/components/niu/__init__.py

```diff
@@ -0,0 +1,128 @@
+"""The NIU integration."""
+from datetime import timedelta
+import logging
+
+from niu import NiuAPIException, NiuCloud
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_BATTERY_CHARGING,
+    ATTR_BATTERY_LEVEL,
+    CONF_SCAN_INTERVAL,
+    CONF_TOKEN,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import DEFAULT_SCAN_INTERVAL, DOMAIN, NIU_COMPONENTS
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up NIU component."""
+
+    hass.data.setdefault(DOMAIN, {})
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up NIU as config entry."""
+
+    config = entry.data
+    username = entry.title
+    token = config[CONF_TOKEN]
+    scan_interval = entry.options.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
```

Comment:
It looks like you didn't implement the [OptionsFlow](https://developers.home-assistant.io/docs/config_entries_options_flow_handler/) to make this value configurable.